lib/schema/utils.js

```diff
@@ -153,10 +156,370 @@ const parseNestedPropertyForResourceAndField = ({ arr, isValueArray }) => {
+      const nestedPayloadProperties = ctx?.instancePath?.split('/')?.slice(1) || [];
+
+      const resourceName = validationContext.getActiveResource();
+      const version = validationContext.getVersion();
+      const validationConfig = validationContext.getValidationConfig();
+      const activeSchema = validationContext.getSchema();
+      const isValueArray = validationContext.getPayloadType() === 'MULTI';
+
+      const { fieldName } = parseNestedPropertyForResourceAndField({
+        arr: nestedPayloadProperties,
+        isValueArray
+      });
+
+      if (typeof data === 'string') {
+        // Process the string to convert it into an array based on the enum definitions.
+        const parsedEnumValues = resolveStringList({
+          string: data,
+          enums: schema,
+          resourceName,
+          fieldName,
+          schema: activeSchema
+        });
+
+        // Validate each item against the schema's enum
+        for (const enumValue of parsedEnumValues) {
+          if (schema.includes(enumValue)) {
+            valid.push(true);
+          } else {
+            // Collect the needed error data if validation fails
+            valid.push(false);
+            let errorMessage;
+            if (validationConfig?.[version]?.[resourceName]?.[fieldName]?.[IGNORE_ENUMS]) {
+              // convert to warning is the failed field is ingored in the validation config
+              errorMessage = {
+                keyword: 'enum',
+                failedItemValue: enumValue,
+                params: {
+                  allowedValues: schema
+                },
+                message: `The following enumerations in the ${fieldName} Field were not advertised. This will fail in Data Dictionary 2.1`,
+                isWarning: true
+              };
+            } else {
+              errorMessage = {
+                keyword: 'enum',
+                failedItemValue: enumValue,
+                params: {
+                  allowedValues: schema
+                },
+                message: 'must be equal to one of the allowed values'
+              };
+            }
+            validate.errors = validate.errors || [];
+            validate.errors.push(errorMessage);
+          }
+        }
+      } else {
+        // Non-string data should not be validated here; let other validations handle that.
+        return true;
+      }
+      return valid.every(Boolean); // Only return true if all validations passed
+    },
+    errors: true // This option tells AJV to use custom errors collected in validate.errors
+  });
+};
+
+const VALIDATION_ERROR_MESSAGES = Object.freeze({
+  NO_CONTEXT_PROPERTY: 'No properties were found in the payload that match "@reso.context"'
+});
+
+const SCHEMA_ERROR_KEYWORDS = {
+  MAX_LENGTH: 'maxLength',
+  ERROR_MESSAGE: 'errorMessage',
+  ENUM: 'enum',
+  ADDITIONAL_PROPERTIES: 'additionalProperties',
+  TYPE: 'type',
+  OBJECT: 'object',
+  NULL: 'null'
+};
+
+/**
+ * TODO
```

Comment:
Perhaps add a comment to describe that this is determining the type so it can be used in a message/string.